File: stand-gradle-plugin/src/main/kotlin/com/ww/gradle/MyTransform.java
package com.ww.gradle;

import com.android.build.api.transform.QualifiedContent;
import com.android.build.api.transform.Transform;
import org.gradle.api.Project

import java.util.Set;

/**
 * @Author weiwei
 * @Date 2022/7/11 18:00
 */
public class MyTransform extends Transform {

    Project project;

    public MyTransform(Project project){
        this.project = project;
    }
    @Override
    public String getName() {
        return null;
    }

    @Override
    public Set<QualifiedContent.ContentType> getInputTypes() {
        return null;
    }

    @Override
    public Set<? super QualifiedContent.Scope> getScopes() {
        return null;
    }

    @Override
    public boolean isIncremental() {
        return false;
    }
}

File: stand-gradle-plugin/src/main/kotlin/com/ww/gradle/MyTransform.kt
package com.ww.gradle

import com.android.build.api.transform.QualifiedContent
import com.android.build.api.transform.Transform
import com.android.build.gradle.internal.pipeline.TransformManager
import org.gradle.api.Project

/**
 * @Author weiwei
 * @Date 2022/7/15 17:51
 */
class MyTransform(private val project: Project) : Transform() {

    companion object {
        private const val TAG = "MyTransform"
    }

    override fun getName(): String? {
        return "TestTransform"
    }

    override fun getInputTypes(): Set<QualifiedContent.ContentType>? {
        return TransformManager.CONTENT_CLASS
    }

    override fun getScopes(): MutableSet<in QualifiedContent.Scope>? {
        return TransformManager.SCOPE_FULL_PROJECT
    }

    override fun isIncremental(): Boolean {
        return false
    }


}